builtin/logical/pki/storage_migrations.go

```diff
@@ -16,55 +16,88 @@ import (
 // and we need to perform it again...
 const latestMigrationVersion = 1
 
-type legacyBundleMigration struct {
+type legacyBundleMigrationLog struct {
 	Hash             string    `json:"hash" structs:"hash" mapstructure:"hash"`
 	Created          time.Time `json:"created" structs:"created" mapstructure:"created"`
 	MigrationVersion int       `json:"migrationVersion" structs:"migrationVersion" mapstructure:"migrationVersion"`
 }
 
-func migrateStorage(ctx context.Context, req *logical.InitializationRequest, logger log.Logger) error {
-	s := req.Storage
-	legacyBundle, err := getLegacyCertBundle(ctx, s)
+type migrationInfo struct {
+	isRequired       bool
+	legacyBundle     *certutil.CertBundle
+	legacyBundleHash string
+	migrationLog     *legacyBundleMigrationLog
+}
+
+func getMigrationInfo(ctx context.Context, s logical.Storage) (migrationInfo, error) {
+	migrationInfo := migrationInfo{
+		isRequired:       false,
+		legacyBundle:     nil,
+		legacyBundleHash: "",
+	}
+	var err error
+
+	migrationInfo.legacyBundle, err = getLegacyCertBundle(ctx, s)
 	if err != nil {
-		return err
+		return migrationInfo, err
```

Comment:
If  this has been deleted (or never created), does it mean we'll err out here?

Reply:
Ah, I think we don't err. Only on other storage issues.

Reply:
Correct, but this comment led me to a missing nil check within fetchCAInfo so thanks 👍